packages/flutter_tools/lib/src/persistent_tool_state.dart

```diff
@@ -96,55 +94,59 @@ class _DefaultPersistentToolState implements PersistentToolState {
   final Config _config;
 
   @override
-  bool get redisplayWelcomeMessage {
-    return _config.getValue(_kRedisplayWelcomeMessage) as bool;
+  bool? get redisplayWelcomeMessage {
+    return _config.getValue(_kRedisplayWelcomeMessage) as bool?;
   }
 
   @override
-  String lastActiveVersion(Channel channel) {
-    final String versionKey = _versionKeyFor(channel);
+  String? lastActiveVersion(Channel channel) {
+    final String? versionKey = _versionKeyFor(channel);
     assert(versionKey != null);
-    return _config.getValue(versionKey) as String;
+    return _config.getValue(versionKey!) as String?;
```

Comment:
Is there any way to mark that the getter returns a nullable but the setter requires a nonnull value?

Reply:
I think that sort of thing is frowned upon.

Reply:
I was frowning the whole time.

Reply:
Now the getters and setters are named something slightly different.